var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -431,7 +433,7 @@ class Trilinos(CMakePackage, CudaPackage, ROCmPackage):
     depends_on("strumpack+shared", when="+strumpack")
     depends_on("suite-sparse", when="+suite-sparse")
     depends_on("superlu-dist", when="+superlu-dist")
-    depends_on("superlu@4.3 +pic", when="+superlu")
+    depends_on("superlu@5.2.2 +pic", when="+superlu")
```

Comment:
```suggestion
    depends_on("superlu@4.3: +pic", when="+superlu")
```
Also would this be valid?  If trilinos can support more than one version it would be best to embed that knowledge in the package

Reply:
The problem is this is not "Spack" valid in the sense that you cannot concretize with version 4.3 in the currently develop version of Spack. Actually this probably means that we want to get rid of the code path that supports 4.3 in Trilinos directly but that's a different level of effort.

Reply:
You can if you set `config:deprecated:true`, or if the users define a `superlu` package that doesn't have 4.3 deprecated.

I would prefer our package recipe be written based on what the code **can** successfully compile against, and allow the spack solver to pick non-deprecated versions. 

An exception to this would be security concerns.

If we had written our `superlu` dependency as a range from the start then the concretization failure wouldn't have occurred when the `superlu` version was deprecated.

Reply:
```suggestion
    depends_on("superlu@4.3: +pic", when="+superlu")
```
Should definitely be this.